File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/HttpEncode.java
package closer.vlllage.com.closer.handler.helpers;

import android.support.annotation.Nullable;

import java.io.UnsupportedEncodingException;
import java.net.URLDecoder;
import java.net.URLEncoder;

import closer.vlllage.com.closer.pool.PoolMember;

public class HttpEncode extends PoolMember {

    private static final String UTF_8 = "UTF-8";

    @Nullable
    public String encode(@Nullable String string) {
        if (string == null) {
            return null;
        }

        try {
            return URLEncoder.encode(string, UTF_8);
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
            return null;
        }
    }

    @Nullable
    public String decode(@Nullable String string) {
        if (string == null) {
            return null;
        }

        try {
            return URLDecoder.decode(string, UTF_8);
        } catch (UnsupportedEncodingException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/HttpEncode.kt
package closer.vlllage.com.closer.handler.helpers

import com.queatz.on.On
import java.io.UnsupportedEncodingException
import java.net.URLDecoder
import java.net.URLEncoder

class HttpEncode constructor(private val on: On) {

    fun encode(string: String?): String? {
        if (string == null) {
            return null
        }

        return try {
            URLEncoder.encode(string, UTF_8)
        } catch (e: UnsupportedEncodingException) {
            e.printStackTrace()
            null
        }

    }

    fun decode(string: String?): String? {
        if (string == null) {
            return null
        }

        return try {
            URLDecoder.decode(string, UTF_8)
        } catch (e: UnsupportedEncodingException) {
            e.printStackTrace()
            null
        }

    }

    companion object {
        private const val UTF_8 = "UTF-8"
    }
}
